agent-control/src/agent_type/runtime_config/on_host/package.rs

```diff
@@ -0,0 +1,207 @@
+        })
+    }
+}
+
+impl Templateable for Download {
+    type Output = rendered::Download;
+    fn template_with(self, variables: &Variables) -> Result<Self::Output, AgentTypeError> {
+        Ok(Self::Output {
+            oci: self.oci.template_with(variables)?,
+        })
+    }
+}
+
+impl Templateable for Oci {
+    type Output = rendered::Oci;
+    fn template_with(self, variables: &Variables) -> Result<Self::Output, AgentTypeError> {
+        let registry = self.registry.template_with(variables)?;
+        let repository = self.repository.template_with(variables)?;
+        let mut version = self.version.template_with(variables)?;
+
+        if !version.is_empty() && !version.starts_with('@') {
+            version = format!(":{}", version);
+        }
+
+        let reference =
+            Reference::from_str(format!("{}/{}{}", registry, repository, version).as_str())
+                .map_err(|err| AgentTypeError::OCIReferenceParsingError(err.to_string()))?;
+
+        Ok(Self::Output { reference })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::agent_type::definition::Variables;
+    use crate::agent_type::runtime_config::templateable_value::TemplateableValue;
+    use crate::agent_type::variable::Variable;
+    use oci_spec::distribution::Reference;
+
+    #[test]
+    fn test_package_type_from_str() {
+        assert_eq!(PackageType::from_str("tar").unwrap(), PackageType::Tar);
+        assert_eq!(PackageType::from_str("tar.gz").unwrap(), PackageType::Tar);
+        assert_eq!(PackageType::from_str("zip").unwrap(), PackageType::Zip);
+        assert!(PackageType::from_str("unsupported").is_err());
+    }
+
+    #[test]
+    fn test_oci_template_only_digest() {
+        let digest = "@sha256:ec5f08ee7be8b557cd1fc5ae1a0ac985e8538da7c93f51a51eff4b277509a723";
+
+        let mut variables = Variables::new();
+        variables.insert(
+            "nr-var:registry".to_string(),
+            Variable::new_final_string_variable("registry.com".to_string()),
+        );
+        variables.insert(
+            "nr-var:repository".to_string(),
+            Variable::new_final_string_variable("repo".to_string()),
+        );
+        variables.insert(
+            "nr-var:version".to_string(),
+            Variable::new_final_string_variable(digest.to_string()),
+        );
+
+        let oci = Oci {
+            registry: TemplateableValue::from_template("${nr-var:registry}".to_string()),
+            repository: TemplateableValue::from_template("${nr-var:repository}".to_string()),
+            version: TemplateableValue::from_template("${nr-var:version}".to_string()),
+        };
+
+        let rendered_oci = oci.template_with(&variables);
+        assert!(rendered_oci.is_ok());
+
+        let expected_reference =
+            Reference::from_str(format!("{}{}", "registry.com/repo", digest).as_str()).unwrap();
+        assert_eq!(rendered_oci.unwrap().reference, expected_reference);
```

Comment:
I'd also try to avoid usign the same implementation `Reference::from_str` in the test and the implementation. If the dependency was updated and, for instance, digest started to be ignored, such behavior change would be unnoted. We could instead check the expected values:

```rust
assert_eq!(rendered_oci.reference.registry(), "registry.com");
// ...
assert_eq!(rendered_oci.reference.digest(), digest);
```